bootable-jar/WFLY-13305_Bootable_Jar.adoc

```diff
@@ -0,0 +1,291 @@
+
+If no galleon layers are specified, then a configuration identical to the default _standalone-microprofile.xml_ is provisioned.
+
+===== CLI script execution
+
+When building the bootable jar, a path to a set of CLI scripts can be provided to update the server configuration packaged in the bootable jar. The CLI scripts are concatenated and executed once the server is provisioned and deployment copied into it (if any).
+Having multiple CLI scripts provides enough flexibility to reuse scripts in different contexts.
+The plugin can be configured with a CLI properties file to resolve properties present in scripts.
+
+===== User application
+
+The war or jar main artifact the maven project is building is deployed and zipped along with the server. 
+The plugin option root-url-path=true|false (true by default) allows to rename the war file to ROOT.war 
+(doesn’t apply to other packaging). NB: The deployment scanner is not required it is even suggested to exclude it, 
+useless in a bootable jar context.
+
+In case the bootable is an “hollow jar”, no deployment is deployed to the server.
+
+===== Hollow jar
+
+No war/jar/ear file is copied into the server deployments. This is controlled by a plugin option.
+The hollow jar doesn’t require the deployment-scanner to be provisioned in order for the deployment to be taken into account at runtime time. At runtime, deployment is copied to content dir, xml configuration is updated with the deployment.
+
+
+==== Execute the bootable jar
+
+This is done thanks to the command: java -jar <jar name>.jar [arguments]
+The detailed execution steps are:
+
+* Unzip the server to a directory
+* Handle arguments
+* Start the server
+* Wait for server end 
+
+Alternatively the maven plugin can be used to run/shutdown the server from maven wildfly-jar:run|start|shutdown.
+
+==== Configure the Bootable jar at runtime
+
+The bootable jar fully relies on the server configuration capabilities. It is not expected to define a new way to configure the server.
+The arguments passed to the bootable jar are composed of bootable specific arguments and server arguments.
+
+Calling  java <JVM and system properties> -jar <bootable jar name>.jar --help dumps the available arguments.
+
+==== Log manager
+
+The bootable jar depends on JBOSS log manager. Logging is configured thanks to the logging subsystem.
+
+==== Shutting down the Bootable jar
+
+The bootable jar process can be shutdown in the following ways:
+
+* Signal handling
+* :shutdown management operation with the following limitations:
+** The restart option is not supported and would be ignored by the server. The process will exit in all cases.
+
+
+==== Read Only standalone.xml
+
+The changes made to the management model are not reflected in the standalone.xml. 
+The server is internally started with --read-only-server-config=standalone.xml
+
+==== Arguments handling
+
+Arguments specific to bootable jar:
```

Comment:
I'd suggest to use a table here for better readability.

Reply:
Done